### DISCUSSION TITLE
Npm audit reports High Security Risk with Package BL - dependency of strapi-plugin-upload

### DISCUSSION THREAD

**Discussion by SorinGFS** [strapi/strapi#7946] [Legacy Topics]
**High**            Remote Memory Exposure                                        
  Package         bl                                                            
  Dependency of   strapi-plugin-upload                                          
  Path            strapi-plugin-upload > sharp > prebuild-install > tar-fs >    
                  tar-stream > bl                                                                                                                       
  More info       https://npmjs.com/advisories/1555

**Comment by derrickmehaffy** [comment]
Per the advisory it only applies to the `consume()` function in which we do not use this.

**Reply by derrickmehaffy** [reply]
Also can you check on the latest version `v3.1.5` as I believe we updated the sharp version in the latest release.

**Reply by SorinGFS** [reply]
Thank you for info. I got that warn when I tried to update to version 3.1.5. For some reason, only 2 strapi dependent packages gets updated when I use `npm update`: strapi-plugin-graphql, and strapi-plugin-documentation, the rest of them are stil 3.1.4. Is this normal behavior or did I use wrong update command?

**Reply by derrickmehaffy** [reply]
You shouldn't use `npm update` nor `npm audit --fix` both are not entirely compatible with Strapi due to our versioning system.

See: https://strapi.io/documentation/v3.x/guides/update-version.html#upgrading-your-dependencies
And always double check there is no migration guide: https://strapi.io/documentation/v3.x/migration-guide/

**Reply by SorinGFS** [reply]
I never use `npm audit --fix`, only `npm audit`, and only when I get some red chars in console

**Reply by SorinGFS** [reply]
I tried `npm install`, same result!

**Reply by SorinGFS** [reply]
I also removed the `package-lock.json`, same!

**Reply by derrickmehaffy** [reply]
I see you marked my response as an answer, did you get it figured out?

**Reply by SorinGFS** [reply]
Nope, I marked the response for initial question, which you answered that strapi does not use that function. The rest is a second question! :)

**Reply by SorinGFS** [reply]
Right now this update matter isn't important to me, I work to build a proxy for multiple strapi app installs in order to avoid the refresh token problem. Almost done!

**Reply by derrickmehaffy** [reply]
Nice!

**Comment by SorinGFS** [comment]
@derrickmehaffy Solved the upgrade problem. Solution was to manually uninstall all dependencies, and intall back one by one. Then `npm run build`.
